src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+    /// qubit wire of the circuit.
+    ///
+    /// # Arguments
+    /// `&mut self` - exposes the Circuit
+    /// `qubits` - qubits used in the quil program
+    /// `instruction` - exposes qubits in a single instruction
+    fn set_qw(&mut self, qubits: &HashSet<Qubit>, instruction: &Instruction) {
+        for program_qubit in qubits {
+            let mut found = false;
+            match &instruction {
+                instruction::Instruction::Gate(gate) => {
+                    for gate_qubit in &gate.qubits {
+                        if program_qubit == gate_qubit {
+                            found = true;
+                        }
+                    }
+
+                    if !found {
+                        match program_qubit {
+                            instruction::Qubit::Fixed(q) => {
+                                self.circuit
+                                    .get_mut(&q)
+                                    .and_then(|wire| wire.empty.insert(self.column, Command::Qw));
+                            }
+                            _ => (),
+                        }
+                    }
+                }
+                _ => (),
+            }
+        }
+    }
+
+    /// Returns a reformatted gate name based on the modifiers used in a single
+    /// instruction line of a quil program or the original name. Gates with
+    /// CONTROLLED modifiers are reformatted such that each CONTROLLED modifier
+    /// prepends a `C` to the beginning of the gate name. For other modifiers
+    /// such as DAGGER, no special reformatting is required.
+    ///
+    /// For example, for an instruction line `CONTROLLED CONTROLLED Y 0 1 2` the
+    /// gate name is reformatted to `CCY` where each C is mapped to an
+    /// associated qubit with the last qubit to the original gate name. For an
+    /// instruction line `DAGGER DAGGER Y 0`, the gate name remains `Y`,
+    /// instead each of the modifiers are added to the wire at the current
+    /// column where it is to be applied using the Command::Super variant.
+    ///
+    /// # Arguments
+    /// `&mut self` - exposes the current column of the Circuit
+    /// `gate` - exposes the modifiers associated with the instruction gate
+    /// `wire` - exposes the wire to be pushed to in Circuit
+    fn set_modifiers(&mut self, gate: &Gate, wire: &mut Wire) -> String {
+        let mut gate_name = gate.name.clone();
+
+        // set modifers
+        if !gate.modifiers.is_empty() {
+            for modifer in &gate.modifiers {
+                match modifer {
+                    instruction::GateModifier::Dagger => {
+                        if let Some(modifiers) = wire.modifiers.get_mut(&self.column) {
+                            modifiers.push("dagger".to_string())
+                        } else {
+                            wire.modifiers
+                                .insert(self.column, vec!["dagger".to_string()]);
+                        }
```

Comment:
Behold, the `entry` API!

This is not necessarily _better_ (there's a potentially unnecessary `clone`), but it's a good API to be familiar with.

```suggestion
                        wire.modifiers.entry(self.column.clone())
                            .and_modify(|modifiers| modifiers.push("dagger".to_string()))
                            .or_insert_with(|| vec!["dagger".to_string()]);
```